pkg/api/validation/validation.go

```diff
@@ -782,9 +782,13 @@ func validateKeyToPath(kp *api.KeyToPath, fldPath *field.Path) field.ErrorList {
 }
 
 func validatePersistentClaimVolumeSource(claim *api.PersistentVolumeClaimVolumeSource, fldPath *field.Path) field.ErrorList {
+	namePath := fldPath.Child("claimName")
+
 	allErrs := field.ErrorList{}
 	if len(claim.ClaimName) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("claimName"), ""))
+		allErrs = append(allErrs, field.Required(namePath, ""))
+	} else {
+		allErrs = append(allErrs, ValidateDNS1123Label(claim.ClaimName, namePath)...)
```

Comment:
We can't restrict the PVC name length.  This will break backwards compatibility.

We had a similar name length issue for gce pds, and the workaround that we did was to truncate the generated name.  But it is prone to name collisions.